ruby_enumerable_methods.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    for item in self do
+      yield(item)
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    for item in self do
+      yield(item, i)
+      i += 1
+    end
+  end
+
+  def my_select
+    select_arr = []
+    for item in self do
+      select_arr << item if yield(item)
+    end
+    select_arr
+  end
+
+  def my_all?
+    pass = false
+    for item in self do
+      pass = yield(item)
+      break unless pass
+    end
+    pass
+  end
```

Comment:
- enumerables my_all? when no block or argument is given returns true when none of the collection members are false or nil
     Failure/Error: expect(true_array.my_all?).to be true_array.all?

- enumerables my_all? when a class is passed as an argument returns true if all of the collection is a member of such class
     Failure/Error: expect(array.my_all?(Integer)).to be array.all?(Integer)

- enumerables my_all? when a Regex is passed as an argument returns true if all of the collection matches the Regex
     Failure/Error: expect(words.my_all?(/d/)).to be words.all?(/d/)

- enumerables my_all? when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern
     Failure/Error: expect(array.my_all?(3)).to be array.all?(3)